uaclient/entitlements/base.py

```diff
@@ -444,14 +554,103 @@ def applicability_status(self) -> "Tuple[ApplicabilityStatus, str]":
         return ApplicabilityStatus.APPLICABLE, ""
 
     @abc.abstractmethod
+    def _perform_disable(self, silent: bool = False) -> bool:
+        """
+        Disable specific entitlement. This should be implemented by subclasses.
+        This method does the actual disable, and does not check can_disable
+        or handle pre_disable or post_disable messaging.
+
+        @return: True on success, False otherwise.
+
+        @param silent: Boolean set True to silence print/log of messages
+
+        @return: True on success, False otherwise.
+        """
+        pass
+
+    def _disable_dependent_services(self):
```

Comment:
The naming of handle_required_services vs _disable_dependent_services gives raises a question about naming. While I'd steer toward naming this handle_dependent_services, it then loses its descriptive name which clarifies what the method actually does. Should handle_required_services be renamed `enable_required_services` instead?